examples/two-columns-layout/src/plugins/productsPlugin.tsx

```diff
@@ -32,15 +32,25 @@ export const productsPlugin: AutocompletePlugin<ProductHit, {}> = {
     return [
       {
         sourceId: 'productsPlugin',
-        getItems() {
+        getItems({ state }) {
+          const previewContext = state.context.preview as ProductsPluginContext;
+
+          const facetFilters = [];
+          if (previewContext?.facetName) {
+            facetFilters.push(
+              `${previewContext.facetName}:${previewContext.facetValue}`
+            );
+          }
+
           return getAlgoliaResults<ProductHit>({
             searchClient,
             queries: [
               {
                 indexName: ALGOLIA_PRODUCTS_INDEX_NAME,
-                query,
+                query: previewContext?.query ?? query,
```

Comment:
Why do you need nullish coalescing here?

Reply:
The `preview` context can be `null`.

Reply:
When the `preview` context is `null`, the current `query` is used to look for matching products.